### ISSUE TITLE
Iteration Persistent Differential Sugar Demand

### ISSUE THREAD

**Issue by wkneh** [WheezePuppet/specstar#76]
To better understand the effects of proto membership on the success of agents, stagger the finite starvation shocks of agents. 

Why? 
    In reality, wealth pooling institutions rely on differential demand of its wealth reserves by its members. Those experiencing success will contribute to the wealth pool, increasing the availability of wealth. Those going through hard times will benefit from this available wealth as they will use this safety net to support them until they become successful once again. 
    The same dynamic can be seen in the simulation, as staggered starvation periods would allow the agents in protos to rely on the contributions of other agents to survive a period of starvation of limited time. The current stage 3 simulates a wholesale starvation event which subverts the role of the proto into a simple equalizing storage container (probably).

How?
Replace the simultaneous starvation period in "stage 3" with staggered starvation shocks of limited length, beginning at a random time for each agent after proto formation.

Results?
This could show that members of a proto survive longer then isolates, as the proto safety net allow the agents to 'pull through' the starvation shock and realize the wealth growth following the starvation shock. If the amount of sugar necessary to support the starving agent is less than the expected future contributions of the agent, the proto group benefits in two ways: the net effect on the proto reserves will be positive, and the average lifespan will increase as the agent in question survived. If instead, that amount is less than the future contributions the effect on the proto reserves will be negative, but the average lifespan will still be impacted positively. Compared to the current simulation, isolates might have a shorter life expectancy as some agents could hit the starvation period early in the simulation (before protracted wealth growth) and not have the necessary sugar reserves to survive. In this context, it is conceivable that proto membership would have an overall positive affect on the success of an agent.

Note: The above logic might only be the case if the expected sugar contribution of a given agent i.e. total wealth growth (ignoring early death) --- loss during starvation shock, is not less then zero. If that sum is larger than zero, the role of the proto to ensure that the wealth growth after the starvation shock is realized in the form of proto contributions, would be most apparent (and positive). Worth investigating for all three cases -,0,+

**Comment by venkatachalapathy** [comment]
Yes. I want to play around with variants of starvation experiments. Test conditions must associated with some specific questions that we want answered about the system. Yes, exploratory questions are the first step; but eventually, we have to focus around a specific question that we want the experimental intervention to answer.

**Comment by divilian** [comment]
I really like this idea, and I also like how you framed it, Will.
